README.md

```diff
@@ -1,71 +1,157 @@
 
 ```sh
-export GOPATH="$HOME/go"
-export PATH="${PATH}:${GOPATH}/bin"
+ko apply -f config/
 ```
 
-### Clone to your machine
+This will create all the prerequisite custom resource definitions and deploy the KCD controller and webhook to your cluster. To verify that the controller and webhook have been correctly deployed, you can use the following command: 
+
+```sh
+kubectl get pod -n knative-serving
+```
 
-1. [Fork this repo](https://help.github.com/articles/fork-a-repo/) to your account
-2. Clone to your machine
+In the printout, if you can find two lines similar to the ones below, then your KCD controller and webhook are correctly deployed:
+
+```
+NAME                                                            READY   STATUS    RESTARTS   AGE
+continuous-delivery-webhook-677cf69bcd-k4lls                    1/1     Running   0          89s
+knative-continuous-delivery-dg8g6-deployment-6d97dd9869-r6hp9   1/2     Running   0          88s
+```
+
+### Applying your Policy or use a built-in Policy
+
+A few built-in policies have been provided in `example-policies/`. You can apply them by doing
 
 ```sh
-mkdir -p ${GOPATH}/src/github.com/googleinterns
-cd ${GOPATH}/src/github.com/googleinterns
-git clone git@github.com:${YOUR_GITHUB_USERNAME}/knative-continuous-delivery.git
-cd knative-continuous-delivery
-git remote add upstream https://github.com/googleinterns/knative-continuous-delivery.git
-git remote set-url --push upstream no_push
+ko apply -f example-policies
 ```
 
-`Upstream` will allow for you to [sync your fork](https://help.github.com/articles/syncing-a-fork/)
+Note that to actually use a built-in policy in a rollout, the relevant Service or Configuration must have the appropriate annotations to specify the name of the policy that it is using.
 
-### Building
+### Deploying your service
 
-`go build ./...`
+Suppose you have a Knative Service `default/my-app` in a file called `my-app.yaml`, then you can simply run the following command to deploy it:
 
-### Running unit tests
+```sh
+kubectl apply -f my-app.yaml
+```
 
-`go test ./...`
+If this is the first Revision of `my-app`, then the behavior is the same as if KCD is not present. However, if this is not the first Revision of `my-app`, and if `my-app.yaml` correctly identifies a rollout policy in the annotations, then KCD will automatically compute the correct routing state for `my-app` and initiate a progressive rollout. You can watch the traffic percentages in real time via the following command:
 
-### Updating dependencies
+```sh
+kubectl get -w route my-app -o yaml
+```
+
+## Common Failures & Diagnostics
+
+### Incorrect traffic behavior
+
+If you see that the rollout percentage for a Revision you just created stays at 0 or jumps to 100 directly, there can be a few potential culprits:
 
-`./hack/update-deps.sh`
+1. You have not applied any rollout policy.
+2. You have not annotated your Service/Configuration with the correct policy name.
```

Comment:
Could there be an issue with a namespace mismatch?

Reply:
Yes, and it is mentioned in the bulletin point right below this line

Reply:
what I'm thinking about is that the `Policy` is not in the same namespace as the `Configuration`. https://github.com/googleinterns/knative-continuous-delivery/blob/97d8aa7d6640e08e8efd6b9684089a5fedcc0dcc/pkg/reconciler/delivery/delivery.go#L146. I think it's a little unclear that policies are namespaced resources and I could see users creating all their policies in one namespace, not realizing that applications in other namespaces cannot access those policies.

We could add a feature that if the policy name contains a slash then it's assumed to be a cross-namespace policy query: `delivery.knative.dev/policy: other-namespace/my-policy` rather than take the namespace from the configuration.

Reply:
Good idea. I have modified the README a little to describe this feature (lines 27-33). I will submit another PR after this to implement it.